src/adt.h

```diff
@@ -37,6 +38,7 @@ int adt_check_header(const void *adt);
 
 static inline int adt_first_property_offset(const void *adt, int offset)
 {
+    (void)(adt);
```

Comment:
This should be UNUSED

Reply:
Sure, but that implies to include utils.h where the macro is defined just for this line. I was expecting this argument be used in near future or removed otherwise.  But I can change this

Reply:
This argument will never be used. It is present for API uniformity, but the way the format works the first property offset is at a constant offset, so no data needs to be read.

Reply:
Got it. I've been seeing the use of the typedefs of `types.h` in the project (u8, u32, etc.). If the UNUSED macro is moved from utils.h to types.h, by including it in adt.h we can also use these typedefs in the adt code and standardize the use of these typedefs. Do you agree to do it and use this same PR?